[6/n][a3][worker] add worker impl + watcher + callbacks

xanderec opened the pull request:
Summary:

worker sets watch on itself

when it gets assigned a task, the data updates to the task node's name

worker then fetches task data from task znode asynchronously

once it gets data, it computes it and creates result znode under the task znode.

Test Plan:

it builds for now, pray it works

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/raydatray/comp-512/pull/157).
* #170
* #169
* #168
* #167
* #166
* #165
* #164
* #163
* #162
* #161
* #160
* #159
* #158
* __->__ #157

raydatray commented on a3/dist/src/main/java/worker/DistWorker.java:
```diff
@@ -0,0 +1,154 @@
+                    resultPath,
+                    resultData,
+                    Ids.OPEN_ACL_UNSAFE,
+                    CreateMode.PERSISTENT
```
why is this persistent?

raydatray reviewed (approved):
mostly makes sense to me

Commit:
[6/n][a3][worker] add worker impl + watcher + callbacks

Summary:

worker sets watch on itself

when it gets assigned a task, the data updates to the task node's name

worker then fetches task data from task znode asynchronously

once it gets data, it computes it and creates result znode under the task znode.

Test Plan:

it builds for now, pray it works
